### PR TITLE
Issue #4157: added support for varargs.

### PR DESCRIPTION
**Pull request by sergileon**
Fixed [https://github.com/checkstyle/checkstyle/issues/4157](https://github.com/checkstyle/checkstyle/issues/4157)

### CODE REVIEW

**Comment by sergileon** [comment]
[report](https://sergileon.github.io/4157/). @romani , I generated the report using the command:
`groovy diff.groovy --localGitRepo /D:\gitEpam\checkstyle --patchBranch whitespacAroundVarargs --patchConfig my_check.xml --listOfProjects projects-for-circle.properties --mode single`
Why in the report are no files with error?

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java [hunk 1]
```diff
@@ -159,7 +159,8 @@
                 // we prefer no spaces on one side or both for these tokens
                 "ARRAY_INIT",
                 // these are covered by GenericWhitespaceCheck
-                "WILDCARD_TYPE", "GENERIC_END", "GENERIC_START").collect(Collectors.toSet()));
+                "WILDCARD_TYPE", "GENERIC_END", "GENERIC_START",
+                "ELLIPSIS").collect(Collectors.toSet()));
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java]: ELLIPSIS is not "covered by GenericWhitespaceCheck"
please put it above this comment.
commend should be "we prefer no space around '...' the same as '[]'"

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java [hunk 2]
```diff
@@ -210,7 +211,7 @@
                 //  allowed via '4.8.3 Arrays'
                 "ARRAY_INIT",
                 // google prefers no spaces on one side or both for these tokens
-                "GENERIC_START", "GENERIC_END", "WILDCARD_TYPE")
+                "GENERIC_START", "GENERIC_END", "WILDCARD_TYPE", "ELLIPSIS")
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java]: comment for "ELLIPSIS" should be " '...' is almost same as '[]' by meaning, google style never define it clearly but in all examples there are no space between type and '...'"

**Comment by rnveach** [comment]
@sergileon 
Please make sure you reply to each point raised. We do not get notifications when commits have been updated, and it will help verify everything we asked is done.

**Pull request merged**